engine/crates/runtime/src/pg.rs

```diff
@@ -17,6 +19,8 @@ pub trait PgTransportFactoryInner {
         name: &str,
         database_definition: &DatabaseDefinition,
     ) -> PgTransportFactoryResult<Box<dyn Transport>>;
+
+    fn fetch_cached(&self, name: &str) -> PgTransportFactoryResult<&dyn Transport>;
```

Comment:
I think you could just change the `try_new` method instead. The `database_definition` method isn't used either in the cloud version: https://github.com/grafbase/api/blob/main/common/grafbase-runtime-cloud/src/postgres.rs#L33. Implementations will have access to the `DatabaseDefinition` when they're created, so wasn't really necessary I think.

Reply:
The difference here is that the other returns a reference and the other owned value.

Reply:
ah good point, didn't notice it. I guess an `Arc` could have been an alternative